Certain decoding error tests fail on big-endian platforms

**Describe the bug**
On big-endian platforms (e.g. `s390x`), several tests fail:

```
=========================== short test summary info ============================
FAILED pynetdicom/tests/test_utils.py::TestSetUID::test_bytes_decoding_error
FAILED pynetdicom/tests/test_utils.py::TestDecodeBytes::test_decoding_error
FAILED pynetdicom/tests/test_utils.py::TestDecodeBytes::test_decoding_error_fallback
==== 3 failed, 2861 passed, 30 skipped, 444 deselected in 834.75s (0:13:54) ====
```

**Expected behavior**

All tests pass.

**Steps To Reproduce**

Run the tests on a big-endian platform.

This may be easier said than done in general. As a Fedora packager, I’m able to do it non-interactively as part of the Fedora Linux RPM build infrastructure, or interactively with a `mock` chroot and `qemu-user-static` emulation.

**Your environment**
Please run the following and paste the output.
```bash
$ python -c "import platform; print(platform.platform())"
Linux-5.16.12-200.fc35.x86_64-s390x-with-glibc2.35.9000
$ python -c "import sys; print('Python ', sys.version)"
Python  3.10.2 (main, Feb  1 2022, 00:00:00) [GCC 12.0.1 20220129 (Red Hat 12.0.1-0)]
$ python -c "import pydicom; print('pydicom ', pydicom.__version__)"
pydicom  2.2.2
$ python -c "import pynetdicom; print('pynetdicom ', pynetdicom.__version__)"
pynetdicom  2.0.1
```

**Further notes**

I will post detailed failing test output in a follow-up comment.

Happily, the problem does not seem to be in the library, but in the byte strings used for testing. The output of the `utf_32` codec depends on the host endianness, and differing bytestrings can produce exceptions with different types and messages than those that the tests expect. I will follow up with a PR that makes these bytestrings host-endian-independent.

--- a/pynetdicom/tests/test_utils.py
+++ b/pynetdicom/tests/test_utils.py
@@ -1,5 +1,6 @@
 """Unit tests for the pynetdicom.utils module."""
 
+from codecs import BOM_UTF32_LE
 from io import BytesIO
 from threading import Thread
 import logging
@@ -230,7 +231,7 @@ class TestSetUID:
 
     def test_bytes_decoding_error(self, caplog):
         """Test invalid bytes raises exception"""
-        b = "1.2.3".encode("utf_32")
+        b = BOM_UTF32_LE + "1.2.3".encode("utf_32_le")
         assert isinstance(b, bytes)
         msg = (
             r"Unable to decode 'FF FE 00 00 31 00 00 00 2E 00 00 00 32 00 00 "
@@ -312,7 +313,7 @@ class TestDecodeBytes:
 
     def test_decoding_error(self, caplog):
         """Test decoding error raises and logs"""
-        b = "1.2.3".encode("utf_32")
+        b = BOM_UTF32_LE + "1.2.3".encode("utf_32_le")
         with caplog.at_level(logging.ERROR, logger="pynetdicom"):
             msg = (
                 r"Unable to decode 'FF FE 00 00 31 00 00 00 2E 00 00 00 32 "
@@ -325,7 +326,7 @@ class TestDecodeBytes:
 
     def test_decoding_error_fallback(self, caplog, utf8):
         """Test decoding error raises and logs"""
-        b = "1.2.3".encode("utf_32")
+        b = BOM_UTF32_LE + "1.2.3".encode("utf_32_le")
         with caplog.at_level(logging.ERROR, logger="pynetdicom"):
             msg = (
                 r"Unable to decode 'FF FE 00 00 31 00 00 00 2E 00 00 00 32 "

